src/bb_excel/core.clj

```diff
@@ -130,127 +100,178 @@
-
-(defn process-row
-  "Process Excel row of data"
-  [shared-strings styles row]
-  (let [row* (->> row
-                  (map (fn [cell] (merge (:attrs cell)
-                                         {:d (-> cell :content last :content last)})))
-                  (map (partial process-cell shared-strings styles)))
-        row-index (get-row-index (first row*))]
-    (into {:_r row-index} (map #(-> [(keyword (:x %)) (:d %)]))
-          row*)))
+      (= cell-type "s")                 (nth shared-strings (parse-long raw-cell-value))
+      (= cell-type "str")               raw-cell-value
+      (= cell-type "inlineStr")         raw-cell-value
+      (= cell-type "b")                 (if (= "1" raw-cell-value) true false)
+      (= cell-type "e")                 (get error-codes raw-cell-value)
+      (= cell-type "n")                 (parse-long raw-cell-value)
+      (style-check cell-attrs styles pcts)    (num2pct raw-cell-value)
+      (style-check cell-attrs styles dates)   (num2date raw-cell-value)
+      (style-check cell-attrs styles times)   (num2time raw-cell-value)
+      :else raw-cell-value)))
 
 (defn- get-cell-text
   "Extract text from cell"
   [cell]
   (->> (xml-seq cell)
-       (filter #(contains? TEXT_T_TAGS (:tag %)))
+       (filter (by-tag :t))
        (mapcat :content)
        (str/join)))
 
 (defn get-shared-strings
   "Get dictionary of all unique strings in the Excel spreadsheet"
   [^ZipFile zipfile]
-  (if-let [wb (.getEntry zipfile "xl/sharedStrings.xml")]
-    (let [ins (.getInputStream zipfile wb)
-          x (parse-str (slurp ins))]
-      (into [] (comp (filter #(contains? TEXT_PART_TAGS (:tag %)))
-                     (map get-cell-text))
-            (xml-seq x)))
+  (if-let [shared-strings-entry (.getEntry zipfile "xl/sharedStrings.xml")]
+    (with-open [shared-strings (.getInputStream zipfile shared-strings-entry)]
+      (let [sst-node (xml/parse shared-strings {:namespace-aware false})]
+        (mapv get-cell-text (:content sst-node))))
     []))
 
 (defn get-styles
-  "Get styles"
   [^ZipFile zipfile]
-  (if-let [wb (.getEntry zipfile "xl/styles.xml")]
-    (let [ins (.getInputStream zipfile wb)
-          x   (parse-str (slurp ins))]
-      (->> (xml-seq x)
-           (filter #((:cellxfs tags) (:tag %)))
-           first
-           :content
-           (filter #((:xf tags) (:tag %)))
-           (mapv (comp :numFmtId :attrs))))
+  (if-let [styles-entry (.getEntry zipfile "xl/styles.xml")]
+    (with-open [styles (.getInputStream zipfile styles-entry)]
+      (let [style-sheet-node (xml/parse styles {:namespace-aware false})
+            cell-xfs-node (->> (:content style-sheet-node)
+                               (find-first (by-tag :cellXfs)))
+            xf-nodes (->> (:content cell-xfs-node)
+                          (filter (by-tag :xf)))]
+        (mapv #(-> % :attrs :numFmtId) xf-nodes)))
     []))
 
+(def ^:const BASE_ROW_INDEX 0)
+(def ^:const START_COLUMN_INDEX 1)
```

Comment:
I'm not sure about this naming, but enough changes in one PR. Maybe need to rethink later.